src/Persistence/Sql/Query.php

```diff
@@ -0,0 +1,339 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\data\Persistence\Sql;
+
+use atk4\data\Exception;
+use atk4\data\Field;
+use atk4\data\FieldSqlExpression;
+use atk4\data\Model;
+use atk4\data\Persistence;
+use atk4\dsql\Expression;
+use atk4\dsql\Expressionable;
+use atk4\dsql\Query as DsqlQuery;
+
+/**
+ * @method Query get()
+ * @method Query getOne()
+ * @method Query limit($limit, $offset = 0)
+ * @method Query order($fields)
+ * @method Query getDebugQuery()
+ * @method Query render()
+ * @method Query mode()
+ */
+class Query implements Persistence\QueryInterface, Expressionable
+{
+    /** @var Model */
+    protected $model;
```

Comment:
it seems we do not need a model here

Reply:
In the query we definitely need the Model